### PR TITLE
Setup initial docker-compose file

### PR DESCRIPTION
**Pull request by pieper126**
## What?

Add docker compose file.

## Why?

This enables us to add integration tests to this repository.

### CODE REVIEW

File: config/mosquitto.conf
Comment (serroba) [inline config/mosquitto.conf:1]: Oh, we should be using a different folder here instead.

The main config here shouldn't be mixed by testing files. This config would be better off in a testing folder instead

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    environment:
+      - CONFIGURATION_PATH=/config/configuration.yaml
+    volumes:
+      - ./config:/config
+    links:
+      - mosquitto
+      - pymodbus-server
+    depends_on:
+      - mosquitto
+      - pymodbus-server
```
Comment (serroba) [inline docker-compose.yaml:46]: New lines please!

**Review by serroba — COMMENTED** [review]
Please review comments before merging

File: config/mosquitto.conf
Comment (pieper126) [reply config/mosquitto.conf:1]: I'll put it in the tests folder.

File: docker-compose.yaml (hunk 1 above)
Comment (pieper126) [reply docker-compose.yaml]: Added the newline.

**Pull request merged**